src/libs/actions/IOU.ts

```diff
@@ -2519,6 +2519,13 @@ function getUpdateMoneyRequestParams(
     let updatedTransaction: OnyxEntry<OnyxTypes.Transaction> = transaction ? TransactionUtils.getUpdatedTransaction(transaction, transactionChanges, isFromExpenseReport) : undefined;
     const transactionDetails = ReportUtils.getTransactionDetails(updatedTransaction);
 
+    if (updatedTransaction?.pendingFields) {
+        pendingFields = {
+            ...pendingFields,
+            ...updatedTransaction?.pendingFields,
+        };
+    }
+
```

Comment:
@paultsimura The problem is `getUpdatedTransaction` function can return another `pendingFields`

Reply:
> Yes, and they will be passed to optimisticData because we use ...updatedTransaction (which includes the new pendingFields) as the first line in the code I've referenced.


Yes, that's correct.